donkeycar/parts/web_controller/web.py

```diff
@@ -194,12 +196,22 @@ def run_threaded(self, img_arr=None, num_records=0, mode=None, recording=None):
             if self.num_records % 10 == 0:
                 changes['num_records'] = self.num_records
 
+        #
+        # get latched button presses then clear button presses
+        # Next iteration will clear press in memory
+        #
+        buttons = self.buttons
+        self.buttons = {}
+        for button in buttons:
+            if buttons[button]:
```

Comment:
Here I'm checking to see if there is a button press.  I'm not checking to see if there is a button.  So if the Button is pressed then buttons[button] is true, so I want to clear it so we only see the button press once.

Reply:
Ah got it, `buttons` is a `dict`, so probably we should try to avoid a duplicated lookup and iterate through `buttons.items()` which returns keys and values together?

Reply:
I made that change.  It reads better now also.  Thanks.